File: 3n+1problem.py
# coding: utf-8
# Rodrigo Farias de Macêdo
def cycleLength(x):
	z = 1
	while(x != 1 ):
		if x%2 == 0:
			x = x/2
		else:
			x = 3*x +1
		z += 1
	return z


entrada = raw_input()
while(entrada != ""):
	entrada = entrada.split()
	x = int(entrada[0])
	y = int(entrada[1])
	max = 0

	for i in range(x, y, 1):
		a = cycleLength(i)
		if a > max:
			max  = a
	print x, y, max
	entrada = raw_input()

File: 3n+1problem.cpp
// Rodrigo Farias de Macêdo

#include <string>
#include <iostream>
#include <cstdio>
#include <sstream>
#include <cctype>
#include <utility>
#include <algorithm> 


using namespace std;

int cycleLength(int x){
	int z = 1;
	while(x != 1 ){
	
		if (x%2 == 0){	
			x = x/2;
		}else{
			x = 3*x +1;
		}
		z += 1;
	}
	return z;
}

int main(){
	int l, r;
	while(scanf("%d %d", &l, &r) != EOF){
		int max = 0;
		
		if (l>r){
			for (int i = r; i <= l; i++){
				int a = cycleLength(i);
				if (a > max){
					max = a;
				}
			}	
		}else{
			for (int i = l; i <= r; i++){
 				int a = cycleLength(i);
				if (a > max){
					max = a;
				}
			}
		}
		printf("%d %d %d\n", l ,r, max);
	}
	return 0;
}
